lstchain/datachecks/dl1_checker.py

```diff
@@ -1436,16 +1436,10 @@ def get_pixel_location(pix_id):
     # The first time we read in the data stored in the io directory:
     infilename = Path(Path(os.path.dirname(__file__)).parent,'io',
                       'LST_pixid_to_cluster.txt')
+    data = np.genfromtxt(infilename, comments='#',dtype='int')
 
-    with open(infilename) as file:
-        for line in file:
-            if line.strip()[0] != '#':
-                pixel_hardware_info.append([])
-        file.seek(0)
-        for line in file:
-            if line[0] == '#':
-                continue
-            data = [int(s) for s in line.split()]
-            pixel_hardware_info[data[0]] = [data[1], data[2], data[3]]
+    pixel_hardware_info.extend([None]*(1 + data[:,0].max()))
```

Comment:
It is a global list, yes. Probably can be done with something like a static variable? (just my ignorance)
The function does what explained in its description: return the "hardware coordinates" of a pixel from its software id

Reply:
Okay, I see. This should be split into two functions. One for reading this information from the file resources into an appropriate in-memory data structure (e.g. a mapping dict or a pandas DataFrame).

And a second function performing the lookup. 
The second function might not be needed if the data structure used directly offers what you need.

Reply:
> Okay, I see. This should be split into two functions. One for reading this information from the file resources into an appropriate in-memory data structure (e.g. a mapping dict or a pandas DataFrame).

But isn't the simple python list fine?  I also do not see the advantage of splitting into two functions.

> 
> And a second function performing the lookup.
> The second function might not be needed if the data structure used directly offers what you need.

I think it does, because the index of the list is directly pixel id.